From: Ben Hutchings <benh@debian.org>
Date:
Subject: icmpv6: Avoid ABI change in 4.9.259
Forwarded: not-needed

Commit 0c5bdc21049f "net: icmp: pass zeroed opts from
icmp{,v6}_ndo_send before sending" changed icmpv6_send() from an
extern exported function to an inline function.  Change it back to
avoid an ABI change.

---
--- a/include/linux/icmpv6.h
+++ b/include/linux/icmpv6.h
@@ -43,10 +43,7 @@ extern int inet6_register_icmp_sender(ip
 extern int inet6_unregister_icmp_sender(ip6_icmp_send_t *fn);
 #endif
 
-static inline void icmpv6_send(struct sk_buff *skb, u8 type, u8 code, __u32 info)
-{
-	__icmpv6_send(skb, type, code, info, IP6CB(skb));
-}
+void icmpv6_send(struct sk_buff *skb, u8 type, u8 code, __u32 info);
 
 int ip6_err_gen_icmpv6_unreach(struct sk_buff *skb, int nhs, int type,
 			       unsigned int data_len);
--- a/net/ipv6/ip6_icmp.c
+++ b/net/ipv6/ip6_icmp.c
@@ -46,6 +46,12 @@ void __icmpv6_send(struct sk_buff *skb,
 EXPORT_SYMBOL(__icmpv6_send);
 #endif
 
+void icmpv6_send(struct sk_buff *skb, u8 type, u8 code, __u32 info)
+{
+	__icmpv6_send(skb, type, code, info, IP6CB(skb));
+}
+EXPORT_SYMBOL(icmpv6_send);
+
 #if IS_ENABLED(CONFIG_NF_NAT)
 #include <net/netfilter/nf_conntrack.h>
 void icmpv6_ndo_send(struct sk_buff *skb_in, u8 type, u8 code, __u32 info)
